### PR TITLE
WMDP-124: Add pydantic, factory-boy, and adjust logging

### PR DESCRIPTION
**Pull request by chouinar**
Note this builds ontop of https://github.com/navapbc/wic-mt-demo-project-mock-api/pull/10

## Ticket

https://wicmtdp.atlassian.net/browse/WMDP-124

## Changes
Added Pydantic including:
* Request/response models
* Environment variable loading

Added a utility for fetching general pieces that most APIs will need.

Added DB factory - not needed yet, but we'll want it when we build the CSV logic.

Adjusted logging to be more configurable, and removed need for always using the decode log script.

## Context for reviewers
This should be a lot of the rest of miscellaneous setup prior to building out the API + adding auth.

The most noteworthy bit is the `ApiContext` I added. In the prior approach, you'd do roughly the following in most APIs:
```
body = connexion.request.json
with app.db_session() as db_session:
    current_user = app.current_user()
    <db operations, selecting by user>
```
In the interest of removing a lot of this boiler plate, especially the slightly confusing connexion request pseudo-global var, I've moved this into an ApiContext class that'll fetch all of that for you. Note that for non-post requests, we'd need to see if the request body is even set (if not, just a minor adjustment to make it not fetch it).

## Testing
Nothing outwardly different for this from prior PRs.

`make clean-volumes init start` will restart everything locally, and you can query the local API.

### CODE REVIEW

File: app/api/util/pydantic_util.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+from pydantic import BaseModel, BaseSettings
+
+import api
+
+env_file = os.path.join(
+    os.path.dirname(os.path.dirname(api.__file__)),
+    "config",
+    "%s.env" % os.getenv("ENVIRONMENT", "local"),
+)
```
Comment (rocketnova) [inline app/api/util/pydantic_util.py:11]: Nice!

File: app/api/util/api_context.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+def api_context_manager() -> Generator[ApiContext, None, None]:
+    """
+    API context manager for working with
+    requests and processing them to the DB.
+
+    Sets up the DB session, current user,
+    and grabs the request body.
+    """
+    with app.db_session() as db_session:
+        body = connexion.request.json
```
Comment (rocketnova) [inline app/api/util/api_context.py:35]: Worth adding a TODO here for checking if the request body is set

File: app/tests/api/db/models/test_factories.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    # DB factory access is disabled from tests unless you add the
+    # 'initialize_factories_session' fixture.
+    with pytest.raises(Exception, match="DB_FACTORIES_DISABLE_DB_ACCESS is set"):
+        EligibilityScreenerFactory.create()
+
+
+def test_eligibility_screener_factory_create(test_db_session, initialize_factories_session):
+    # Build doesn't use the DB
+
+    # Build sets the values
```
Comment (rocketnova) [inline app/tests/api/db/models/test_factories.py]: Update these comments to describe `create()`?

File: app/tests/api/db/test_migrations.py [hunk 4]
```diff
@@ -60,8 +57,6 @@ def test_db_schema_non_session(monkeypatch):
 
 
 def test_db_setup_via_alembic_migration(test_db_schema_non_session):
-    # Change directory location so the relative script_location in alembic config works.
-    os.chdir(Path(__file__).parent.parent)
```
Comment (rocketnova) [inline app/tests/api/db/test_migrations.py:64]: Why don't we want to test this?

**Review by rocketnova — APPROVED** [review]
Looks great! Thank you! I particularly like the `ApiContext` and the logging changes.

File: app/tests/api/db/test_migrations.py (hunk 4 above)
Comment (chouinar) [reply app/tests/api/db/test_migrations.py:64]: It was making the test print the path for some reason + removing it didn't break anything so I think it's not relevant (note: this was copied verbatim, might be an old alembic issue).

Comment (chouinar) [reply app/tests/api/db/test_migrations.py:64]: No idea what was going on, tinkered with it and it stopped printing.

I don't think we need it in either case (it works both ways). The bit I copied from grabbed the config differently by some hacky file paths, so likely just irrelevant in this approach.

I did add a bit to the test to validate via log statements that the migration occurred.

**Pull request merged**